integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
  * #L%
  */
 
-import com.thinkbiganalytics.spark.DataSet;
 import com.thinkbiganalytics.spark.SparkContextService;
 
+import com.thinkbiganalytics.spark.dataquality.util.FlowAttributes;
+import com.thinkbiganalytics.spark.dataquality.util.MissingAttributeException;
+import com.thinkbiganalytics.spark.dataquality.util.MissingRuleException;
+import com.thinkbiganalytics.spark.dataquality.util.DataQualityConstants;
+import com.thinkbiganalytics.spark.dataquality.output.DataQualityRow;
+import com.thinkbiganalytics.spark.dataquality.output.DataQualityWriter;
+
+import com.thinkbiganalytics.spark.dataquality.rule.DataQualityRule;
+import com.thinkbiganalytics.spark.dataquality.rule.InvalidRowPercentRuleImpl;
+import com.thinkbiganalytics.spark.dataquality.rule.InvalidRowTotalCountRuleImpl;
+import com.thinkbiganalytics.spark.dataquality.rule.RowCountRuleImpl;
+import com.thinkbiganalytics.spark.dataquality.rule.SourceToFeedCountRuleImpl;
+
 import org.apache.commons.lang3.StringUtils;
 import org.apache.spark.SparkContext;
-import org.apache.spark.sql.Row;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.DataFrame;
 import org.apache.spark.sql.hive.HiveContext;
+import org.json.simple.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 /**
- * Perform data quality checks
+ * This class performs the Data Quality Checks. This is done by executing various Data Quality rules
+ * and using the passed in JSON file which contains all the necessary attributes. The results of all
+ * the tests are stored into a feed_dataquality hive table<br>
+ * <br>
+ * This class takes in one argument which is the path of the JSON file which contains all the
+ * flowfile attributes<br>
+ * <br>
+ * Failure is considered to be if any of the rules failed.<br>
```

Comment:
Please reword to make it more clear. E.g. "If any of the rules fail, the overall check is considered a failure."